workflow_scripts/update_onnx_hub_manifest.py

```diff
@@ -0,0 +1,122 @@
+manifest_object = []
+
+all_tables = []
+for markdown_file in markdown_files:
+    with open(markdown_file, "r") as f:
+        parsed = markdown.markdown(f.read(), extensions=["markdown.extensions.tables"])
+        soup = bs4.BeautifulSoup(parsed, "html.parser")
+        for table in soup.find_all("table"):
+            parsed = parse_html(table)
+            if all(col in parsed.columns.values for col in ["Model", "Download", "Opset version", "ONNX version"]):
+                parsed["source_file"] = markdown_file
+                all_tables.append(parsed)
+            else:
+                print("Unrecognized table columns in file {}: {}".format(markdown_file, parsed.columns.values))
+
+df = pd.concat(all_tables, axis=0)
+normalize_name = {
+    'Download': "model_path",
+    'Download (with sample test data)': 'model_with_data_path',
+}
+
+top_level_fields = [
+    "model", "model_path", "opset_version", "onnx_version"
+]
+
+
+def prep_name(col):
+    if col in normalize_name:
+        col = normalize_name[col]
+    col = col.rstrip()
+    prepped_col = col.replace(" ", "_").lower()
+    if prepped_col in top_level_fields:
+        return prepped_col
+    else:
+        return col
+
+
+renamed = df.rename(columns={col: prep_name(col) for col in df.columns.values})
+metadata_fields = [f for f in renamed.columns.values if f not in top_level_fields]
+
+
+def get_file_info(row, feild):
+    source_dir = split(row["source_file"])[0]
+    model_file = row[feild].contents[0].attrs['href']
+    rel_path = os.sep.join(join(source_dir, model_file).split(os.sep)[1:])
+    with open(join("..", rel_path), "rb") as f:
+        bytes = f.read()
+        sha256 = hashlib.sha256(bytes).hexdigest()
+    return {
+        feild: rel_path,
+        feild.replace("_path", "") + "_sha": sha256,
+        feild.replace("_path", "") + "_bytes": len(bytes)}
+
+
+output = []
+for i, row in renamed.iterrows():
+
+    model_info = get_file_info(row, "model_path")
+    model_path = model_info.pop("model_path")
+    metadata = model_info
+
+    try:
+        for k, v in get_file_info(row, "model_with_data_path").items():
+            metadata[k] = v
+    except AttributeError as e:
+        print("no model_with_data in file {}".format(row["source_file"]))
+
+    try:
+        opset = int(row["opset_version"].contents[0])
+    except ValueError:
+        print("malformed opset {} in {}".format(row["opset_version"].contents[0], row["source_file"]))
+        continue
+
+    if len(row["model"].contents) > 0:
+        output.append({
+            "model": row["model"].contents[0],
+            "model_path": model_path,
+            "onnx_version": row["onnx_version"].contents[0],
+            "opset_version": int(row["opset_version"].contents[0]),
+            "metadata": metadata})
+    else:
+        print("Missing model in {}".format(row["source_file"]))
+
+with open(join("..", "ONNX_HUB_MANIFEST.json"), "w+") as f:
+    json.dump(output, f, indent=4)

```

Comment:
Add top-level `repositories` list.
```suggestion
manifest_object = {
    "repositories": [
        {"name": "onnx/models/master", "type": "github", "owner": "onnx", "repo": "models", "ref": "master"}
    ],
    "models": output,
}

with open(join("..", "ONNX_HUB_MANIFEST.json"), "w+") as f:
    json.dump(manifest_object, f, indent=4)
```

Reply:
> Add top-level `repositories` list.

One issue i forsee is the inclusion of the "master" in this as the client can point to any version of this by commit hash and branch name. I don't think the client will need any additional info about the repository but let me know what you have in mind for this info

Reply:
Discussed offline. Feel free to reject this change.